### PR TITLE
[SYCL-MLIR][NFCI]: Cleanup

### PR DESCRIPTION
**Pull request by etiotto**
This PR cleans up the code a bit by:
   - making member data private where possible
   - avoid using declaration in header file
   - avoid using namespace std
   - remove dead code

### CODE REVIEW

File: polygeist/tools/cgeist/Lib/clang-mlir.cc [hunk 1]
```diff
@@ -161,14 +163,15 @@ void MLIRScanner::initSupportedFunctions() {
...
-  if (ShowAST) {
-    llvm::errs() << "Emitting fn: " << function.getName() << "\n";
-    llvm::errs() << *fd << "\n";
-  }
+  function = func;
+  EmittingFunctionDecl = fd;
+
+  if (ShowAST)
+    llvm::dbgs() << "Emitting fn: " << function.getName() << "\n"
+                 << *fd << "\n";
```
Comment (victor-eds) [inline polygeist/tools/cgeist/Lib/clang-mlir.cc:174]: `llvm::errs` is used for this in the rest of the file, maybe we should stick to it.

File: polygeist/tools/cgeist/Lib/clang-mlir.cc [hunk 2]
```diff
@@ -5003,30 +5011,31 @@ mlir::func::FuncOp MLIRASTConsumer::GetOrCreateMLIRFunction(
...
       continue;
+
+    LLVM_DEBUG(llvm::dbgs() << "\n-- FUNCTION BEING EMITTED: "
+                            << FD->getNameAsString() << " --\n\n";);
+
     done.insert(name);
     MLIRScanner ms(*this, module, deviceModule, LTInfo);
-    auto Function = GetOrCreateMLIRFunction(FD, true);
-    ms.init(Function, FD);
+    func::FuncOp function = GetOrCreateMLIRFunction(FD, true);
```
Comment (victor-eds) [inline polygeist/tools/cgeist/Lib/clang-mlir.cc:5033]: Same as above

File: polygeist/tools/cgeist/Lib/clang-mlir.h [hunk 3]
```diff
@@ -42,17 +42,15 @@
 #include "clang/../../lib/CodeGen/CodeGenModule.h"
 #include "clang/AST/Mangle.h"
 
-using namespace clang;
-using namespace mlir;
```
Comment (victor-eds) [inline polygeist/tools/cgeist/Lib/clang-mlir.h:46]: I was very much against this, also. Nice.

**Review by victor-eds — DISMISSED** [review]
I think this changes help the code, but my concern is whether this will be detrimental in the future if we want to rebase polygeist. What do you think?

**Review by etiotto — COMMENTED** [review]
Addressed some of the comments. I realize this PR makes merging upstram polygeist more challenging, IMO at some point we should contribute changes back to the polygeist project. 
Attempting to keep the code in sync is prob. futile right now.

File: polygeist/tools/cgeist/Lib/CGCall.cc [hunk 4]
```diff
@@ -1,6 +1,6 @@
 // Copyright (C) Codeplay Software Limited
 
-//===--- CGCall.cpp - Encapsulate calling convention details --------------===//
+//===--- CGCall.cc - Encapsulate calling convention details --------------===//
```
Comment (whitneywhtsang) [inline polygeist/tools/cgeist/Lib/CGCall.cc:3]:
```suggestion
//===--- CGCall.cc - Encapsulate calling convention details ---------------===//
```

Comment (etiotto) [reply polygeist/tools/cgeist/Lib/CGCall.cc]: done

**Review by whitneywhtsang — APPROVED** [review]

**Pull request merged**